lib/livebook/file_system/s3/client.ex

```diff
@@ -259,21 +259,28 @@ defmodule Livebook.FileSystem.S3.Client do
   end
 
   defp sign_headers(file_system, method, url, headers, body \\ nil) do
+    credentials = S3.credentials(file_system)
```

Comment:
The other place where we want to get credentials is here:

https://github.com/livebook-dev/livebook/blob/56734420a07c8a1061aa97d283f13bc36ef92eb7/lib/livebook/session.ex#L2662-L2672